src/utils/healthCheck/utils.spec.ts

```diff
@@ -29,19 +30,94 @@ describe('utils/healthCheck', () => {
       expect(state.value).toBe(true);
-      expect(await state.healthChecker()).toBe(true);
+      expect(await state.healthChecker(CONTEXT)).toBe(true);
 
       // it('becomes unhealthy')
       state.set(false);
 
       expect(state.value).toBe(false);
-      expect(await state.healthChecker()).toBe(false);
+      expect(await state.healthChecker(CONTEXT)).toBe(false);
+    })
+    it('can change states', async () => {
+      // it(defaults to onChange logging strategy)
+      telemetryMock.expects('error').once();
+      const healthCheckStateOptions: HealthCheckStateOptions = {
+        tag: 'productSync'
+      }
+      const state = new HealthCheckState(healthCheckStateOptions);
+
+      // it('is assumed healthy')
+      expect(state.value).toBe(true);
+      expect(await state.healthChecker(CONTEXT)).toBe(true);
+
+      // it('becomes unhealthy')
+      state.set(false);
+
+      expect(state.value).toBe(false);
+      expect(await state.healthChecker(CONTEXT)).toBe(false);
     });
+
+    describe("loggingStrategy", () => {
+      it ("has 'never' set", async () => {
+        telemetryMock.expects('error').never();
+        const healthCheckStateOptions: HealthCheckStateOptions = {
+          tag: 'productSync',
+          loggingStrategy: 'never',
+        }
+        const state = new HealthCheckState(healthCheckStateOptions);
+        state.set(false);
+        //it (switches state to unhealthy)
+        const result = await state.healthChecker(CONTEXT);
+        expect(result).toBe(false);
+
+        // it (should never report to telemetry no matter how many times called)
+        await state.healthChecker(CONTEXT);
+        await state.healthChecker(CONTEXT);
+      })
+      it ("has 'always' set", async () => {
+        telemetryMock.expects('error').thrice();
+        const healthCheckStateOptions: HealthCheckStateOptions = {
+          tag: 'productSync',
+          loggingStrategy: 'always',
+        }
+        const state = new HealthCheckState(healthCheckStateOptions);
+        //it (switches state to unhealthy)
+        state.set(false);
+
+        //it (calls the healthChecker three times)
+        await state.healthChecker(CONTEXT);
+        await state.healthChecker(CONTEXT);
+        await state.healthChecker(CONTEXT);
+
+      })
+      it ("defaults to 'onChange'", async () => {
+        const healthCheckStateOptions: HealthCheckStateOptions = {
+          tag: 'productSync',
+          loggingStrategy: 'onChange',
+        }
+        const state = new HealthCheckState(healthCheckStateOptions);
+        //it (switches state to unhealthy)
+        state.set(false);
+
+        telemetryMock.expects('error').once();
+        await state.healthChecker(CONTEXT);
+
+        //it (should not log another telmetry error)
+        await state.healthChecker(CONTEXT);
+        //it (switches state to healthy)
+        state.set(true);
+
+        telemetryMock.expects('info').once()
+        await state.healthChecker(CONTEXT);
```

Comment:
continuing with your line of thought here,

```suggestion
        await state.healthChecker(CONTEXT);

        //it (should not log another telmetry info)
        await state.healthChecker(CONTEXT);
```